src/lib/sifp/sss_sifp_parser.c

```diff
@@ -434,7 +434,7 @@ sss_sifp_parse_array(sss_sifp_ctx *ctx,
     if (ret != SSS_SIFP_OK) {
         if (attr->type == SSS_SIFP_ATTR_TYPE_STRING && attr->data.str != NULL) {
             for (unsigned int i = 0;
-                 attr->data.str[i] != NULL && i < attr->num_values;
+                 i < attr->num_values && attr->data.str[i] != NULL;
```

Comment:
Matter of taste, of course, but IMO using parentheses here would improve readability.

Reply:
Do you mean `(i < attr->num_values) && (attr->data.str[i] != NULL)`?

Reply:
I have no strong opinion for either option. @pbrezina as you are already reviewing this PR, which one do you prefer?

Reply:
I have no strong opinion either, but it would be easier to read with parentheses.